### Student Last Name: Flynn
### Student First Name: Joseph


Spring 2020 Programming for Analytics CIS 3120 PTRB[26809] (Baruch College) 
HW#1 Web Scrapping

## Question#1:
### Complete the Python program below to display the name of the most recently added dataset and collect the total number of datasets currently listed on the URL provided below.
Note: You may need to install **cssselect** module

In [1]:
!pip install cssselect

  Using cached cssselect-1.1.0-py2.py3-none-any.whl (16 kB)


In [7]:
from lxml import html
import requests
response = requests.get('http://catalog.data.gov/dataset?q=&sort=metadata_created+desc')

#html.fromstring converts the http response into an html element
doc = html.fromstring(response.text)
#a specifc element is then selected with cssselect(). In this case, its the content under the h3 head, under class of dataset-heading.
#The .text_content coverts the element to text
title = doc.cssselect('h3.dataset-heading')[0].text_content()


print("The name of the most recently added dataset on data.gov:")
#strips the title and prints it
print(title.strip())

response1 = requests.get('http://www.data.gov/')

#html.fromstring converts the http response into an html element
doc_gov = html.fromstring(response1.text)
#a specifc element is then selected with cssselect(). Then it selects the element at samll a and the first index.
link_gov = doc_gov.cssselect('small a')[0]

print("Number of datasets currently listed on data.gov:")
#changes specific element to text before printing
print(link_gov.text)


The name of the most recently added dataset on data.gov:
PREDICT Emerging Pandemic Threats Project Data Asset (China - Restricted Data)
Number of datasets currently listed on data.gov:
337,304 datasets


**OUTPUT TEXT: **
The name of the most recently added dataset on data.gov:
Canterbury, New Zealand 2018
Number of datasets currently listed on data.gov:
224,654 datasets

## Question#2:
### Complete the Python program below to download and display the content below.

In [3]:
import requests

# response = requests.get("https://en.wikipedia.org/robots.txt")
response = requests.get("https://en.wikipedia.org/wiki/MediaWiki:Robots.txt")
#converts http response to text
test = response.text#TO DO -- Complete the Code
print("robots.txt for http://www.wikipedia.org/")
print("===================================================")
#prints out html code as text
print(test)


#these two were my attempts to make the html look like bellow


#from urllib.request import urlopen
#from bs4 import BeautifulSoup

#takes link given
#html = urlopen("https://en.wikipedia.org/wiki/MediaWiki:Robots.txt")
#parses link in beautiful soup
#soup = BeautifulSoup(html, 'html.parser')
#uses prettyify to indent the html properly
#prettyHTML = soup.prettify()
#prints prettier html
#print(prettyHTML)

#The difference between https://en.wikipedia.org/robots.txt and https://en.wikipedia.org/wiki/MediaWiki:Robots.txt
#is that https://en.wikipedia.org/robots.txt is just a text file store on the website
#while the https://en.wikipedia.org/wiki/MediaWiki:Robots.txt is the text file as a source on the webpage
#So, you'll get the text allowing with the webpages html


robots.txt for http://www.wikipedia.org/
<!DOCTYPE html>
<html class="client-nojs" lang="en" dir="ltr">
<head>
<meta charset="UTF-8"/>
<title>MediaWiki:Robots.txt - Wikipedia</title>
<script>document.documentElement.className="client-js";RLCONF={"wgBreakFrames":false,"wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgRequestId":"d8ff5018-bd4c-4d00-9f5f-354b6e806df4","wgCSPNonce":false,"wgCanonicalNamespace":"MediaWiki","wgCanonicalSpecialPageName":false,"wgNamespaceNumber":8,"wgPageName":"MediaWiki:Robots.txt","wgTitle":"Robots.txt","wgCurRevisionId":1025811479,"wgRevisionId":1025811479,"wgArticleId":19292575,"wgIsArticle":true,"wgIsRedirect":false,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":[],"wgPageContentLanguage":"en","wgPageContentModel":"wikitext","wgRelevantPageName":"MediaWi

** OUTPUT TEXT: **

robots.txt for http://www.wikipedia.org/
===================================================
<!DOCTYPE html>
<html class="client-nojs" lang="en" dir="ltr">
<head>
<meta charset="UTF-8"/>
<title>MediaWiki:Robots.txt - Wikipedia</title>
<script>document.documentElement.className="client-js";RLCONF={"wgBreakFrames":!1,"wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgRequestId":"YCUlb7z4EqR-V8ltlKtvgwAAANM","wgCSPNonce":!1,"wgCanonicalNamespace":"MediaWiki","wgCanonicalSpecialPageName":!1,"wgNamespaceNumber":8,"wgPageName":"MediaWiki:Robots.txt","wgTitle":"Robots.txt","wgCurRevisionId":978017859,"wgRevisionId":978017859,"wgArticleId":19292575,"wgIsArticle":!0,"wgIsRedirect":!1,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":[],"wgPageContentLanguage":"en","wgPageContentModel":"wikitext","wgRelevantPageName":"MediaWiki:Robots.txt","wgRelevantArticleId":19292575,"wgIsProbablyEditable":!1,"wgRelevantPageIsProbablyEditable":!1,"wgRestrictionEdit":[],"wgRestrictionMove":[],"wgMediaViewerOnClick":!0,
"wgMediaViewerEnabledByDefault":!0,"wgPopupsReferencePreviews":!1,"wgPopupsConflictsWithNavPopupGadget":!1,"wgPopupsConflictsWithRefTooltipsGadget":!0,"wgVisualEditor":{"pageLanguageCode":"en","pageLanguageDir":"ltr","pageVariantFallbacks":"en"},"wgMFDisplayWikibaseDescriptions":{"search":!0,"nearby":!0,"watchlist":!0,"tagline":!1},"wgWMESchemaEditAttemptStepOversample":!1,"wgULSCurrentAutonym":"English","wgNoticeProject":"wikipedia","wgCentralAuthMobileDomain":!1,"wgEditSubmitButtonLabelPublish":!0,"wgULSPosition":"interlanguage"};RLSTATE={"ext.globalCssJs.user.styles":"ready","site.styles":"ready","noscript":"ready","user.styles":"ready","ext.globalCssJs.user":"ready","user":"ready","user.options":"loading","skins.vector.styles.legacy":"ready","ext.visualEditor.desktopArticleTarget.noscript":"ready","ext.uls.interlanguage":"ready","ext.wikimediaBadges":"ready"};RLPAGEMODULES=["site","mediawiki.page.ready","skins.vector.legacy.js",
"ext.gadget.ReferenceTooltips","ext.gadget.charinsert","ext.gadget.extra-toolbar-buttons","ext.gadget.refToolbar","ext.gadget.switcher","ext.centralauth.centralautologin","ext.popups","ext.visualEditor.desktopArticleTarget.init","ext.visualEditor.targetLoader","ext.eventLogging","ext.wikimediaEvents","ext.navigationTiming","ext.uls.compactlinks","ext.uls.interface","ext.cx.eventlogging.campaigns","ext.centralNotice.geoIP","ext.centralNotice.startUp"];</script>
<script>(RLQ=window.RLQ||[]).push(function(){mw.loader.implement("user.options@1hzgi",function($,jQuery,require,module){/*@nomin*/mw.user.tokens.set({"patrolToken":"+\\","watchToken":"+\\","csrfToken":"+\\"});
});});</script>
<link rel="stylesheet" href="/w/load.php?lang=en&amp;modules=ext.uls.interlanguage%7Cext.visualEditor.desktopArticleTarget.noscript%7Cext.wikimediaBadges%7Cskins.vector.styles.legacy&amp;only=styles&amp;skin=vector"/>
<script async="" src="/w/load.php?lang=en&amp;modules=startup&amp;only=scripts&amp;raw=1&amp;skin=vector"></script>
<meta name="ResourceLoaderDynamicStyles" content=""/>
<link rel="stylesheet" href="/w/load.php?lang=en&amp;modules=site.styles&amp;only=styles&amp;skin=vector"/>
<meta name="generator" content="MediaWiki 1.36.0-wmf.27"/>
<meta name="referrer" content="origin"/>
<meta name="referrer" content="origin-when-crossorigin"/>
<meta name="referrer" content="origin-when-cross-origin"/>
<link rel="alternate" media="only screen and (max-width: 720px)" href="//en.m.wikipedia.org/wiki/MediaWiki:Robots.txt"/>
<link rel="apple-touch-icon" href="/static/apple-touch/wikipedia.png"/>
<link rel="shortcut icon" href="/static/favicon/wikipedia.ico"/>
<link rel="search" type="application/opensearchdescription+xml" href="/w/opensearch_desc.php" title="Wikipedia (en)"/>
<link rel="EditURI" type="application/rsd+xml" href="//en.wikipedia.org/w/api.php?action=rsd"/>
<link rel="license" href="//creativecommons.org/licenses/by-sa/3.0/"/>
<link rel="canonical" href="https://en.wikipedia.org/wiki/MediaWiki:Robots.txt"/>
<link rel="dns-prefetch" href="//login.wikimedia.org"/>
<link rel="dns-prefetch" href="//meta.wikimedia.org" />
</head>
<body class="mediawiki ltr sitedir-ltr mw-hide-empty-elt ns-8 ns-subject page-MediaWiki_Robots_txt rootpage-MediaWiki_Robots_txt skin-vector action-view skin-vector-legacy"><div id="mw-page-base" class="noprint"></div>
<div id="mw-head-base" class="noprint"></div>
<div id="content" class="mw-body" role="main">
	<a id="top"></a>
	<div id="siteNotice" class="mw-body-content"><!-- CentralNotice --></div>
	<div class="mw-indicators mw-body-content">
	</div>
	<h1 id="firstHeading" class="firstHeading" lang="en">MediaWiki:Robots.txt</h1>
	<div id="bodyContent" class="mw-body-content">
		<div id="siteSub" class="noprint">From Wikipedia, the free encyclopedia</div>
		<div id="contentSub"></div>
		<div id="contentSub2"></div>
		
		<div id="jump-to-nav"></div>
		<a class="mw-jump-link" href="#mw-head">Jump to navigation</a>
		<a class="mw-jump-link" href="#searchInput">Jump to search</a>
		<div id="mw-content-text" lang="en" dir="ltr" class="mw-content-ltr"><div class="mw-parser-output"><p> # </p><pre>#
# Localisable part of robots.txt for en.wikipedia.org
#
# Edit at https://en.wikipedia.org/w/index.php?title=MediaWiki:Robots.txt&amp;action=edit
# Don't add newlines here. All rules set here are active for every user-agent.
#
# Please check any changes using a syntax validator such as http://tool.motoricerca.info/robots-checker.phtml
# Enter https://en.wikipedia.org/robots.txt as the URL to check.
#
# https://phabricator.wikimedia.org/T16075
Disallow: /wiki/MediaWiki:Spam-blacklist
Disallow: /wiki/MediaWiki%3ASpam-blacklist
Disallow: /wiki/MediaWiki_talk:Spam-blacklist
Disallow: /wiki/MediaWiki_talk%3ASpam-blacklist
Disallow: /wiki/Wikipedia:WikiProject_Spam
Disallow: /wiki/Wikipedia_talk:WikiProject_Spam
#
# Folks get annoyed when XfD discussions end up the number 1 google hit for
# their name. 
# https://phabricator.wikimedia.org/T16075
Disallow: /wiki/Wikipedia:Articles_for_deletion
Disallow: /wiki/Wikipedia%3AArticles_for_deletion
Disallow: /wiki/Wikipedia:Votes_for_deletion
Disallow: /wiki/Wikipedia%3AVotes_for_deletion
Disallow: /wiki/Wikipedia:Pages_for_deletion
Disallow: /wiki/Wikipedia%3APages_for_deletion
Disallow: /wiki/Wikipedia:Miscellany_for_deletion
Disallow: /wiki/Wikipedia%3AMiscellany_for_deletion
Disallow: /wiki/Wikipedia:Miscellaneous_deletion
Disallow: /wiki/Wikipedia%3AMiscellaneous_deletion
Disallow: /wiki/Wikipedia:Categories_for_discussion
Disallow: /wiki/Wikipedia%3ACategories_for_discussion
Disallow: /wiki/Wikipedia:Templates_for_deletion
Disallow: /wiki/Wikipedia%3ATemplates_for_deletion
Disallow: /wiki/Wikipedia:Redirects_for_discussion
Disallow: /wiki/Wikipedia%3ARedirects_for_discussion
Disallow: /wiki/Wikipedia:Deletion_review
Disallow: /wiki/Wikipedia%3ADeletion_review
Disallow: /wiki/Wikipedia:WikiProject_Deletion_sorting
Disallow: /wiki/Wikipedia%3AWikiProject_Deletion_sorting
Disallow: /wiki/Wikipedia:Files_for_deletion
Disallow: /wiki/Wikipedia%3AFiles_for_deletion
Disallow: /wiki/Wikipedia:Files_for_discussion
Disallow: /wiki/Wikipedia%3AFiles_for_discussion
Disallow: /wiki/Wikipedia:Possibly_unfree_files
Disallow: /wiki/Wikipedia%3APossibly_unfree_files
#
# https://phabricator.wikimedia.org/T12288
Disallow: /wiki/Wikipedia_talk:Articles_for_deletion
Disallow: /wiki/Wikipedia_talk%3AArticles_for_deletion
Disallow: /wiki/Wikipedia_talk:Votes_for_deletion
Disallow: /wiki/Wikipedia_talk%3AVotes_for_deletion
Disallow: /wiki/Wikipedia_talk:Pages_for_deletion
Disallow: /wiki/Wikipedia_talk%3APages_for_deletion
Disallow: /wiki/Wikipedia_talk:Miscellany_for_deletion
Disallow: /wiki/Wikipedia_talk%3AMiscellany_for_deletion
Disallow: /wiki/Wikipedia_talk:Miscellaneous_deletion
Disallow: /wiki/Wikipedia_talk%3AMiscellaneous_deletion
Disallow: /wiki/Wikipedia_talk:Templates_for_deletion
Disallow: /wiki/Wikipedia_talk%3ATemplates_for_deletion
Disallow: /wiki/Wikipedia_talk:Categories_for_discussion
Disallow: /wiki/Wikipedia_talk%3ACategories_for_discussion
Disallow: /wiki/Wikipedia_talk:Deletion_review
Disallow: /wiki/Wikipedia_talk%3ADeletion_review
Disallow: /wiki/Wikipedia_talk:WikiProject_Deletion_sorting
Disallow: /wiki/Wikipedia_talk%3AWikiProject_Deletion_sorting
Disallow: /wiki/Wikipedia_talk:Files_for_deletion
Disallow: /wiki/Wikipedia_talk%3AFiles_for_deletion
Disallow: /wiki/Wikipedia_talk:Files_for_discussion
Disallow: /wiki/Wikipedia_talk%3AFiles_for_discussion
Disallow: /wiki/Wikipedia_talk:Possibly_unfree_files
Disallow: /wiki/Wikipedia_talk%3APossibly_unfree_files
#
Disallow: /wiki/Wikipedia:Copyright_problems
Disallow: /wiki/Wikipedia%3ACopyright_problems
Disallow: /wiki/Wikipedia_talk:Copyright_problems
Disallow: /wiki/Wikipedia_talk%3ACopyright_problems
Disallow: /wiki/Wikipedia:Suspected_copyright_violations
Disallow: /wiki/Wikipedia%3ASuspected_copyright_violations
Disallow: /wiki/Wikipedia_talk:Suspected_copyright_violations
Disallow: /wiki/Wikipedia_talk%3ASuspected_copyright_violations
Disallow: /wiki/Wikipedia:Contributor_copyright_investigations
Disallow: /wiki/Wikipedia%3AContributor_copyright_investigations
Disallow: /wiki/Wikipedia:Contributor_copyright_investigations
Disallow: /wiki/Wikipedia%3AContributor_copyright_investigations
Disallow: /wiki/Wikipedia_talk:Contributor_copyright_investigations
Disallow: /wiki/Wikipedia_talk%3AContributor_copyright_investigations
Disallow: /wiki/Wikipedia_talk:Contributor_copyright_investigations
Disallow: /wiki/Wikipedia_talk%3AContributor_copyright_investigations
Disallow: /wiki/Wikipedia:Protected_titles
Disallow: /wiki/Wikipedia%3AProtected_titles
Disallow: /wiki/Wikipedia_talk:Protected_titles
Disallow: /wiki/Wikipedia_talk%3AProtected_titles
Disallow: /wiki/Wikipedia:Articles_for_creation
Disallow: /wiki/Wikipedia%3AArticles_for_creation
Disallow: /wiki/Wikipedia_talk:Articles_for_creation
Disallow: /wiki/Wikipedia_talk%3AArticles_for_creation
Disallow: /wiki/Wikipedia_talk:Article_wizard
Disallow: /wiki/Wikipedia_talk%3AArticle_wizard
#
# https://phabricator.wikimedia.org/T13261
Disallow: /wiki/Wikipedia:Requests_for_arbitration
Disallow: /wiki/Wikipedia%3ARequests_for_arbitration
Disallow: /wiki/Wikipedia_talk:Requests_for_arbitration
Disallow: /wiki/Wikipedia_talk%3ARequests_for_arbitration
Disallow: /wiki/Wikipedia:Requests_for_comment
Disallow: /wiki/Wikipedia%3ARequests_for_comment
Disallow: /wiki/Wikipedia_talk:Requests_for_comment
Disallow: /wiki/Wikipedia_talk%3ARequests_for_comment
Disallow: /wiki/Wikipedia:Requests_for_adminship
Disallow: /wiki/Wikipedia%3ARequests_for_adminship
Disallow: /wiki/Wikipedia_talk:Requests_for_adminship
Disallow: /wiki/Wikipedia_talk%3ARequests_for_adminship
#
# https://phabricator.wikimedia.org/T14111
Disallow: /wiki/Wikipedia:Requests_for_checkuser
Disallow: /wiki/Wikipedia%3ARequests_for_checkuser
Disallow: /wiki/Wikipedia_talk:Requests_for_checkuser
Disallow: /wiki/Wikipedia_talk%3ARequests_for_checkuser
#
# https://phabricator.wikimedia.org/T15398
Disallow: /wiki/Wikipedia:WikiProject_Spam
Disallow: /wiki/Wikipedia%3AWikiProject_Spam
#
# https://phabricator.wikimedia.org/T16793
Disallow: /wiki/Wikipedia:Changing_username
Disallow: /wiki/Wikipedia%3AChanging_username
Disallow: /wiki/Wikipedia:Changing_username
Disallow: /wiki/Wikipedia%3AChanging_username
Disallow: /wiki/Wikipedia_talk:Changing_username
Disallow: /wiki/Wikipedia_talk%3AChanging_username
Disallow: /wiki/Wikipedia_talk:Changing_username
Disallow: /wiki/Wikipedia_talk%3AChanging_username
#
Disallow: /wiki/Wikipedia:Administrators%27_noticeboard
Disallow: /wiki/Wikipedia%3AAdministrators%27_noticeboard
Disallow: /wiki/Wikipedia_talk:Administrators%27_noticeboard
Disallow: /wiki/Wikipedia_talk%3AAdministrators%27_noticeboard
Disallow: /wiki/Wikipedia:Community_sanction_noticeboard
Disallow: /wiki/Wikipedia%3ACommunity_sanction_noticeboard
Disallow: /wiki/Wikipedia_talk:Community_sanction_noticeboard
Disallow: /wiki/Wikipedia_talk%3ACommunity_sanction_noticeboard
Disallow: /wiki/Wikipedia:Bureaucrats%27_noticeboard
Disallow: /wiki/Wikipedia%3ABureaucrats%27_noticeboard
Disallow: /wiki/Wikipedia_talk:Bureaucrats%27_noticeboard
Disallow: /wiki/Wikipedia_talk%3ABureaucrats%27_noticeboard
#
Disallow: /wiki/Wikipedia:Sockpuppet_investigations
Disallow: /wiki/Wikipedia%3ASockpuppet_investigations
Disallow: /wiki/Wikipedia_talk:Sockpuppet_investigations
Disallow: /wiki/Wikipedia_talk%3ASockpuppet_investigations
#
Disallow: /wiki/Wikipedia:Neutral_point_of_view/Noticeboard
Disallow: /wiki/Wikipedia%3ANeutral_point_of_view/Noticeboard
Disallow: /wiki/Wikipedia_talk:Neutral_point_of_view/Noticeboard
Disallow: /wiki/Wikipedia_talk%3ANeutral_point_of_view/Noticeboard
#
Disallow: /wiki/Wikipedia:No_original_research/noticeboard
Disallow: /wiki/Wikipedia%3ANo_original_research/noticeboard
Disallow: /wiki/Wikipedia_talk:No_original_research/noticeboard
Disallow: /wiki/Wikipedia_talk%3ANo_original_research/noticeboard
#
Disallow: /wiki/Wikipedia:Fringe_theories/Noticeboard
Disallow: /wiki/Wikipedia%3AFringe_theories/Noticeboard
Disallow: /wiki/Wikipedia_talk:Fringe_theories/Noticeboard
Disallow: /wiki/Wikipedia_talk%3AFringe_theories/Noticeboard
#
Disallow: /wiki/Wikipedia:Conflict_of_interest/Noticeboard
Disallow: /wiki/Wikipedia%3AConflict_of_interest/Noticeboard
Disallow: /wiki/Wikipedia_talk:Conflict_of_interest/Noticeboard
Disallow: /wiki/Wikipedia_talk%3AConflict_of_interest/Noticeboard
#
Disallow: /wiki/Wikipedia:Long-term_abuse
Disallow: /wiki/Wikipedia%3ALong-term_abuse
Disallow: /wiki/Wikipedia_talk:Long-term_abuse
Disallow: /wiki/Wikipedia_talk%3ALong-term_abuse
Disallow: /wiki/Wikipedia:Long_term_abuse
Disallow: /wiki/Wikipedia%3ALong_term_abuse
Disallow: /wiki/Wikipedia_talk:Long_term_abuse
Disallow: /wiki/Wikipedia_talk%3ALong_term_abuse
#
Disallow: /wiki/Wikipedia:Wikiquette_assistance
Disallow: /wiki/Wikipedia%3AWikiquette_assistance
#
Disallow: /wiki/Wikipedia:Abuse_reports
Disallow: /wiki/Wikipedia%3AAbuse_reports
Disallow: /wiki/Wikipedia_talk:Abuse_reports
Disallow: /wiki/Wikipedia_talk%3AAbuse_reports
Disallow: /wiki/Wikipedia:Abuse_response
Disallow: /wiki/Wikipedia%3AAbuse_response
Disallow: /wiki/Wikipedia_talk:Abuse_response
Disallow: /wiki/Wikipedia_talk%3AAbuse_response
#
Disallow: /wiki/Wikipedia:Reliable_sources/Noticeboard
Disallow: /wiki/Wikipedia%3AReliable_sources/Noticeboard
Disallow: /wiki/Wikipedia_talk:Reliable_sources/Noticeboard
Disallow: /wiki/Wikipedia_talk%3AReliable_sources/Noticeboard
#
Disallow: /wiki/Wikipedia:Suspected_sock_puppets
Disallow: /wiki/Wikipedia%3ASuspected_sock_puppets
Disallow: /wiki/Wikipedia_talk:Suspected_sock_puppets
Disallow: /wiki/Wikipedia_talk%3ASuspected_sock_puppets
#
Disallow: /wiki/Wikipedia:Biographies_of_living_persons/Noticeboard
Disallow: /wiki/Wikipedia%3ABiographies_of_living_persons/Noticeboard
Disallow: /wiki/Wikipedia_talk:Biographies_of_living_persons/Noticeboard
Disallow: /wiki/Wikipedia_talk%3ABiographies_of_living_persons/Noticeboard
Disallow: /wiki/Wikipedia:Biographies_of_living_persons%2FNoticeboard
Disallow: /wiki/Wikipedia%3ABiographies_of_living_persons%2FNoticeboard
Disallow: /wiki/Wikipedia_talk:Biographies_of_living_persons%2FNoticeboard
Disallow: /wiki/Wikipedia_talk%3ABiographies_of_living_persons%2FNoticeboard
#
Disallow: /wiki/Wikipedia:Content_noticeboard
Disallow: /wiki/Wikipedia%3AContent_noticeboard
Disallow: /wiki/Wikipedia_talk:Content_noticeboard
Disallow: /wiki/Wikipedia_talk%3AContent_noticeboard
#
Disallow: /wiki/Template:Editnotices
Disallow: /wiki/Template%3AEditnotices
#
Disallow: /wiki/Wikipedia:Arbitration
Disallow: /wiki/Wikipedia%3AArbitration
Disallow: /wiki/Wikipedia_talk:Arbitration
Disallow: /wiki/Wikipedia_talk%3AArbitration
#
Disallow: /wiki/Wikipedia:Arbitration_Committee
Disallow: /wiki/Wikipedia%3AArbitration_Committee
Disallow: /wiki/Wikipedia_talk:Arbitration_Committee
Disallow: /wiki/Wikipedia_talk%3AArbitration_Committee
#
Disallow: /wiki/Wikipedia:Arbitration_Committee_Elections
Disallow: /wiki/Wikipedia%3AArbitration_Committee_Elections
Disallow: /wiki/Wikipedia_talk:Arbitration_Committee_Elections
Disallow: /wiki/Wikipedia_talk%3AArbitration_Committee_Elections
#
Disallow: /wiki/Wikipedia:Mediation_Committee
Disallow: /wiki/Wikipedia%3AMediation_Committee
Disallow: /wiki/Wikipedia_talk:Mediation_Committee
Disallow: /wiki/Wikipedia_talk%3AMediation_Committee
#
Disallow: /wiki/Wikipedia:Mediation_Cabal/Cases
Disallow: /wiki/Wikipedia%3AMediation_Cabal/Cases
#
Disallow: /wiki/Wikipedia:Requests_for_bureaucratship
Disallow: /wiki/Wikipedia%3ARequests_for_bureaucratship
Disallow: /wiki/Wikipedia_talk:Requests_for_bureaucratship
Disallow: /wiki/Wikipedia_talk%3ARequests_for_bureaucratship
#
Disallow: /wiki/Wikipedia:Administrator_review
Disallow: /wiki/Wikipedia%3AAdministrator_review
Disallow: /wiki/Wikipedia_talk:Administrator_review
Disallow: /wiki/Wikipedia_talk%3AAdministrator_review
#
Disallow: /wiki/Wikipedia:Editor_review
Disallow: /wiki/Wikipedia%3AEditor_review
Disallow: /wiki/Wikipedia_talk:Editor_review
Disallow: /wiki/Wikipedia_talk%3AEditor_review
#
Disallow: /wiki/Wikipedia:Article_Incubator
Disallow: /wiki/Wikipedia%3AArticle_Incubator
Disallow: /wiki/Wikipedia_talk:Article_Incubator
Disallow: /wiki/Wikipedia_talk%3AArticle_Incubator
#
Disallow: /wiki/Category:Noindexed_pages
Disallow: /wiki/Category%3ANoindexed_pages
#
# User sandboxes for modules and Template Styles are placed in these subpages for testing
#
Disallow: /wiki/Module:Sandbox
Disallow: /wiki/Module%3ASandbox
Disallow: /wiki/Template:TemplateStyles_sandbox
Disallow: /wiki/Template%3ATemplateStyles_sandbox
#
# </pre>
<!-- 
NewPP limit report
Parsed by mw1322
Cached time: 20210211123911
Cache expiry: 2592000
Dynamic content: false
Complications: []
CPU time usage: 0.005 seconds
Real time usage: 0.006 seconds
Preprocessor visited node count: 4/1000000
Post‐expand include size: 0/2097152 bytes
Template argument size: 0/2097152 bytes
Highest expansion depth: 2/40
Expensive parser function count: 0/500
Unstrip recursion depth: 0/20
Unstrip post‐expand size: 12549/5000000 bytes
Number of Wikibase entities loaded: 0/400
-->
<!--
Transclusion expansion time report (%,ms,calls,template)
100.00%    0.000      1 -total
-->

<!-- Saved in parser cache with key enwiki:pcache:idhash:19292575-0!canonical and timestamp 20210211123911 and revision id 978017859. Serialized with JSON.
 -->
</div><noscript><img src="//en.wikipedia.org/wiki/Special:CentralAutoLogin/start?type=1x1" alt="" title="" width="1" height="1" style="border: none; position: absolute;" /></noscript>
<div class="printfooter">Retrieved from "<a dir="ltr" href="https://en.wikipedia.org/w/index.php?title=MediaWiki:Robots.txt&amp;oldid=978017859">https://en.wikipedia.org/w/index.php?title=MediaWiki:Robots.txt&amp;oldid=978017859</a>"</div></div>
		<div id="catlinks" class="catlinks catlinks-allhidden" data-mw="interface"></div>
	</div>
</div>
<div id='mw-data-after-content'>
	<div class="read-more-container"></div>
</div>

<div id="mw-navigation">
	<h2>Navigation menu</h2>
	<div id="mw-head">
		<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-personal" class="mw-portlet mw-portlet-personal vector-menu" aria-labelledby="p-personal-label" role="navigation" 
	 >
	<h3 id="p-personal-label">
		<span>Personal tools</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"><li id="pt-anonuserpage">Not logged in</li><li id="pt-anontalk"><a href="/wiki/Special:MyTalk" title="Discussion about edits from this IP address [n]" accesskey="n">Talk</a></li><li id="pt-anoncontribs"><a href="/wiki/Special:MyContributions" title="A list of edits made from this IP address [y]" accesskey="y">Contributions</a></li><li id="pt-createaccount"><a href="/w/index.php?title=Special:CreateAccount&amp;returnto=MediaWiki%3ARobots.txt" title="You are encouraged to create an account and log in; however, it is not mandatory">Create account</a></li><li id="pt-login"><a href="/w/index.php?title=Special:UserLogin&amp;returnto=MediaWiki%3ARobots.txt" title="You&#039;re encouraged to log in; however, it&#039;s not mandatory. [o]" accesskey="o">Log in</a></li></ul>
		
	</div>
</nav>

		<div id="left-navigation">
			<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-namespaces" class="mw-portlet mw-portlet-namespaces vector-menu vector-menu-tabs" aria-labelledby="p-namespaces-label" role="navigation" 
	 >
	<h3 id="p-namespaces-label">
		<span>Namespaces</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"><li id="ca-nstab-mediawiki" class="selected"><a href="/wiki/MediaWiki:Robots.txt" title="View the system message [c]" accesskey="c">Interface page</a></li><li id="ca-talk"><a href="/wiki/MediaWiki_talk:Robots.txt" rel="discussion" title="Discuss improvements to the content page [t]" accesskey="t">Talk</a></li></ul>
		
	</div>
</nav>

			<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-variants" class="mw-portlet mw-portlet-variants emptyPortlet vector-menu vector-menu-dropdown" aria-labelledby="p-variants-label" role="navigation" 
	 >
	<input type="checkbox" class="vector-menu-checkbox" aria-labelledby="p-variants-label" />
	<h3 id="p-variants-label">
		<span>Variants</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"></ul>
		
	</div>
</nav>

		</div>
		<div id="right-navigation">
			<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-views" class="mw-portlet mw-portlet-views vector-menu vector-menu-tabs" aria-labelledby="p-views-label" role="navigation" 
	 >
	<h3 id="p-views-label">
		<span>Views</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"><li id="ca-view" class="selected"><a href="/wiki/MediaWiki:Robots.txt">Read</a></li><li id="ca-viewsource"><a href="/w/index.php?title=MediaWiki:Robots.txt&amp;action=edit" title="This page is protected.&#10;You can view its source [e]" accesskey="e">View source</a></li><li id="ca-history"><a href="/w/index.php?title=MediaWiki:Robots.txt&amp;action=history" title="Past revisions of this page [h]" accesskey="h">View history</a></li></ul>
		
	</div>
</nav>

			<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-cactions" class="mw-portlet mw-portlet-cactions emptyPortlet vector-menu vector-menu-dropdown" aria-labelledby="p-cactions-label" role="navigation" 
	 >
	<input type="checkbox" class="vector-menu-checkbox" aria-labelledby="p-cactions-label" />
	<h3 id="p-cactions-label">
		<span>More</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"></ul>
		
	</div>
</nav>

			<div id="p-search" role="search">
	<h3 >
		<label for="searchInput">Search</label>
	</h3>
	<form action="/w/index.php" id="searchform">
		<div id="simpleSearch" data-search-loc="header-navigation">
			<input type="search" name="search" placeholder="Search Wikipedia" autocapitalize="sentences" title="Search Wikipedia [f]" accesskey="f" id="searchInput"/>
			<input type="hidden" name="title" value="Special:Search">
			<input type="submit" name="fulltext" value="Search" title="Search Wikipedia for this text" id="mw-searchButton" class="searchButton mw-fallbackSearchButton"/>
			<input type="submit" name="go" value="Go" title="Go to a page with this exact name if it exists" id="searchButton" class="searchButton"/>
		</div>
	</form>
</div>

		</div>
	</div>
	
<div id="mw-panel">
	<div id="p-logo" role="banner">
		<a class="mw-wiki-logo" href="/wiki/Main_Page"
			title="Visit the main page"></a>
	</div>
	<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-navigation" class="mw-portlet mw-portlet-navigation vector-menu vector-menu-portal portal" aria-labelledby="p-navigation-label" role="navigation" 
	 >
	<h3 id="p-navigation-label">
		<span>Navigation</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"><li id="n-mainpage-description"><a href="/wiki/Main_Page" title="Visit the main page [z]" accesskey="z">Main page</a></li><li id="n-contents"><a href="/wiki/Wikipedia:Contents" title="Guides to browsing Wikipedia">Contents</a></li><li id="n-currentevents"><a href="/wiki/Portal:Current_events" title="Articles related to current events">Current events</a></li><li id="n-randompage"><a href="/wiki/Special:Random" title="Visit a randomly selected article [x]" accesskey="x">Random article</a></li><li id="n-aboutsite"><a href="/wiki/Wikipedia:About" title="Learn about Wikipedia and how it works">About Wikipedia</a></li><li id="n-contactpage"><a href="//en.wikipedia.org/wiki/Wikipedia:Contact_us" title="How to contact Wikipedia">Contact us</a></li><li id="n-sitesupport"><a href="https://donate.wikimedia.org/wiki/Special:FundraiserRedirector?utm_source=donate&amp;utm_medium=sidebar&amp;utm_campaign=C13_en.wikipedia.org&amp;uselang=en" title="Support us by donating to the Wikimedia Foundation">Donate</a></li></ul>
		
	</div>
</nav>

	<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-interaction" class="mw-portlet mw-portlet-interaction vector-menu vector-menu-portal portal" aria-labelledby="p-interaction-label" role="navigation" 
	 >
	<h3 id="p-interaction-label">
		<span>Contribute</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"><li id="n-help"><a href="/wiki/Help:Contents" title="Guidance on how to use and edit Wikipedia">Help</a></li><li id="n-introduction"><a href="/wiki/Help:Introduction" title="Learn how to edit Wikipedia">Learn to edit</a></li><li id="n-portal"><a href="/wiki/Wikipedia:Community_portal" title="The hub for editors">Community portal</a></li><li id="n-recentchanges"><a href="/wiki/Special:RecentChanges" title="A list of recent changes to Wikipedia [r]" accesskey="r">Recent changes</a></li><li id="n-upload"><a href="/wiki/Wikipedia:File_Upload_Wizard" title="Add images or other media for use on Wikipedia">Upload file</a></li></ul>
		
	</div>
</nav>
<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-tb" class="mw-portlet mw-portlet-tb vector-menu vector-menu-portal portal" aria-labelledby="p-tb-label" role="navigation" 
	 >
	<h3 id="p-tb-label">
		<span>Tools</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"><li id="t-whatlinkshere"><a href="/wiki/Special:WhatLinksHere/MediaWiki:Robots.txt" title="List of all English Wikipedia pages containing links to this page [j]" accesskey="j">What links here</a></li><li id="t-recentchangeslinked"><a href="/wiki/Special:RecentChangesLinked/MediaWiki:Robots.txt" rel="nofollow" title="Recent changes in pages linked from this page [k]" accesskey="k">Related changes</a></li><li id="t-upload"><a href="/wiki/Wikipedia:File_Upload_Wizard" title="Upload files [u]" accesskey="u">Upload file</a></li><li id="t-specialpages"><a href="/wiki/Special:SpecialPages" title="A list of all special pages [q]" accesskey="q">Special pages</a></li><li id="t-permalink"><a href="/w/index.php?title=MediaWiki:Robots.txt&amp;oldid=978017859" title="Permanent link to this revision of this page">Permanent link</a></li><li id="t-info"><a href="/w/index.php?title=MediaWiki:Robots.txt&amp;action=info" title="More information about this page">Page information</a></li></ul>
		
	</div>
</nav>
<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-coll-print_export" class="mw-portlet mw-portlet-coll-print_export vector-menu vector-menu-portal portal" aria-labelledby="p-coll-print_export-label" role="navigation" 
	 >
	<h3 id="p-coll-print_export-label">
		<span>Print/export</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"><li id="coll-download-as-rl"><a href="/w/index.php?title=Special:DownloadAsPdf&amp;page=MediaWiki%3ARobots.txt&amp;action=show-download-screen" title="Download this page as a PDF file">Download as PDF</a></li><li id="t-print"><a href="/w/index.php?title=MediaWiki:Robots.txt&amp;printable=yes" title="Printable version of this page [p]" accesskey="p">Printable version</a></li></ul>
		
	</div>
</nav>

	<!-- Please do not use role attribute as CSS selector, it is deprecated. -->
<nav id="p-lang" class="mw-portlet mw-portlet-lang vector-menu vector-menu-portal portal" aria-labelledby="p-lang-label" role="navigation" 
	 >
	<h3 id="p-lang-label">
		<span>Languages</span>
	</h3>
	<div class="vector-menu-content">
		<ul class="vector-menu-content-list"></ul>
		<div class="after-portlet after-portlet-lang"><span class="uls-after-portlet-link"></span></div>
	</div>
</nav>

</div>

</div>
<footer id="footer" class="mw-footer" role="contentinfo" >
	<ul id="footer-info" >
	<li id="footer-info-lastmod"> This page was last edited on 12 September 2020, at 11:16<span class="anonymous-show">&#160;(UTC)</span>.</li>
	<li id="footer-info-copyright">Text is available under the <a rel="license" href="//en.wikipedia.org/wiki/Wikipedia:Text_of_Creative_Commons_Attribution-ShareAlike_3.0_Unported_License">Creative Commons Attribution-ShareAlike License</a><a rel="license" href="//creativecommons.org/licenses/by-sa/3.0/" style="display:none;"></a>;
additional terms may apply.  By using this site, you agree to the <a href="//foundation.wikimedia.org/wiki/Terms_of_Use">Terms of Use</a> and <a href="//foundation.wikimedia.org/wiki/Privacy_policy">Privacy Policy</a>. Wikipedia® is a registered trademark of the <a href="//www.wikimediafoundation.org/">Wikimedia Foundation, Inc.</a>, a non-profit organization.</li>
</ul>

	<ul id="footer-places" >
	<li id="footer-places-privacy"><a href="https://foundation.wikimedia.org/wiki/Privacy_policy" class="extiw" title="wmf:Privacy policy">Privacy policy</a></li>
	<li id="footer-places-about"><a href="/wiki/Wikipedia:About" title="Wikipedia:About">About Wikipedia</a></li>
	<li id="footer-places-disclaimer"><a href="/wiki/Wikipedia:General_disclaimer" title="Wikipedia:General disclaimer">Disclaimers</a></li>
	<li id="footer-places-contact"><a href="//en.wikipedia.org/wiki/Wikipedia:Contact_us">Contact Wikipedia</a></li>
	<li id="footer-places-mobileview"><a href="//en.m.wikipedia.org/w/index.php?title=MediaWiki:Robots.txt&amp;mobileaction=toggle_view_mobile" class="noprint stopMobileRedirectToggle">Mobile view</a></li>
	<li id="footer-places-developers"><a href="https://www.mediawiki.org/wiki/Special:MyLanguage/How_to_contribute">Developers</a></li>
	<li id="footer-places-statslink"><a href="https://stats.wikimedia.org/#/en.wikipedia.org">Statistics</a></li>
	<li id="footer-places-cookiestatement"><a href="https://foundation.wikimedia.org/wiki/Cookie_statement">Cookie statement</a></li>
</ul>

	<ul id="footer-icons" class="noprint">
	<li id="footer-copyrightico"><a href="https://wikimediafoundation.org/"><img src="/static/images/footer/wikimedia-button.png" srcset="/static/images/footer/wikimedia-button-1.5x.png 1.5x, /static/images/footer/wikimedia-button-2x.png 2x" width="88" height="31" alt="Wikimedia Foundation" loading="lazy" /></a></li>
	<li id="footer-poweredbyico"><a href="https://www.mediawiki.org/"><img src="/static/images/footer/poweredby_mediawiki_88x31.png" alt="Powered by MediaWiki" srcset="/static/images/footer/poweredby_mediawiki_132x47.png 1.5x, /static/images/footer/poweredby_mediawiki_176x62.png 2x" width="88" height="31" loading="lazy"/></a></li>
</ul>

	<div style="clear: both;"></div>
</footer>


<script>(RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgPageParseReport":{"limitreport":{"cputime":"0.005","walltime":"0.006","ppvisitednodes":{"value":4,"limit":1000000},"postexpandincludesize":{"value":0,"limit":2097152},"templateargumentsize":{"value":0,"limit":2097152},"expansiondepth":{"value":2,"limit":40},"expensivefunctioncount":{"value":0,"limit":500},"unstrip-depth":{"value":0,"limit":20},"unstrip-size":{"value":12549,"limit":5000000},"entityaccesscount":{"value":0,"limit":400},"timingprofile":["100.00%    0.000      1 -total"]},"cachereport":{"origin":"mw1322","timestamp":"20210211123911","ttl":2592000,"transientcontent":false}}});mw.config.set({"wgBackendResponseTime":186,"wgHostname":"mw1322"});});</script>
</body></html>

## Question#3:
### Complete the Python program below to extract and display all the header tags from the URL below.
Note: HTML headers are 'h1', 'h2','h3','h4','h5','h6'


In [4]:
from urllib.request import urlopen
from bs4 import BeautifulSoup

#saves data from url
html = urlopen('https://blogs.baruch.cuny.edu/abuk/')
#turns data to html.parser
bs =  BeautifulSoup(html, 'html.parser') 
#setup empty list to save headers to later
titles = []

#loop that finds headers 1 to 6
for found in bs.find_all(['h1', 'h2','h3','h4','h5','h6']):
    #takes found header and stores it in list named titles
    titles.append(found)

#prints title list to have each header spaced by one line
print('List all the header tags :', *titles, sep='\n\n')


List all the header tags :

<h1 class="site-title"><a href="https://blogs.baruch.cuny.edu/abuk/" rel="home">my Teaching Info</a></h1>

<h2 class="entry-title"><a href="https://blogs.baruch.cuny.edu/abuk/%resume%/" rel="bookmark">My Teaching</a></h2>

<h1>Dr. Abu Kamruzzaman, PhD</h1>

<h5>Assistant Professor (Adj.)<br/>
Paul H. Chook Department of Information Systems and Statistics<br/>
Zicklin School of Business<br/>
Baruch College / CUNY<br/>
Box B11-220; One Bernard Baruch Way, New York, NY 10010<br/>
Dept. URL: <a href="https://zicklin.baruch.cuny.edu/faculty-research/academic-departments/information-technology-statistics/" rel="noopener noreferrer" target="_blank">https://zicklin.baruch.cuny.edu/faculty-research/academic-departments/information-technology-statistics/</a></h5>

<h5>Assistant Professor (Adj.)<br/>
Math &amp; Computer Science<br/>
School of Arts &amp; Sciences<br/>
Academic Core Room 2C07<br/>
York College, CUNY<br/>
94 – 20 Guy R. Brewer Blvd.<br/>
Jamaica, NY 11451

**OUTPUT TEXT:**

List all the header tags :

<h1 class="site-title"><a href="https://blogs.baruch.cuny.edu/abuk/" rel="home">my Teaching Info</a></h1>

<h2 class="entry-title"><a href="https://blogs.baruch.cuny.edu/abuk/%resume%/" rel="bookmark">My Teaching</a></h2>

<h1>Dr. Abu Kamruzzaman, PhD</h1>

<h5>Assistant Professor (Adj.)<br/>
Paul H. Chook Department of Information Systems and Statistics<br/>
Zicklin School of Business<br/>
Baruch College / CUNY<br/>
Box B11-220; One Bernard Baruch Way, New York, NY 10010<br/>
Dept. URL: <a href="https://zicklin.baruch.cuny.edu/faculty-research/academic-departments/information-technology-statistics/" rel="noopener noreferrer" target="_blank">https://zicklin.baruch.cuny.edu/faculty-research/academic-departments/information-technology-statistics/</a></h5>

<h5>Assistant Professor (Adj.)<br/>
Math &amp; Computer Science<br/>
School of Arts &amp; Sciences<br/>
Academic Core Room 2C07<br/>
York College, CUNY<br/>
94 – 20 Guy R. Brewer Blvd.<br/>
Jamaica, NY 11451<br/>
Dept. URL: <a href="https://www.york.cuny.edu/academics/departments/mathematics-and-computer-science" rel="noopener noreferrer" target="_blank">https://www.york.cuny.edu/academics/departments/mathematics-and-computer-science</a></h5>

<h5>Member of Doctoral Dissertations for Computer Science PhD<br/>
Program URL: <a href="https://www.pace.edu/seidenberg/phd-cs" rel="noopener noreferrer" style="font-size: 16px;font-weight: 400" target="_blank">https://www.pace.edu/seidenberg/phd-cs</a><br/>
Seidenberg School of CSIS at Pace University<br/>
861 Bedford Road, Pleasantville NY 10570-9913</h5>

<h5>Member of Doctoral Dissertations for Doctor of Professional Studies (DPS) in Computing<br/>
Program URL: <a href="https://www.pace.edu/seidenberg/dps" rel="noopener noreferrer" style="font-size: 16px;font-weight: 400" target="_blank">https://www.pace.edu/seidenberg/dps</a><br/>
Seidenberg School of CSIS at Pace University<br/>
861 Bedford Road, Pleasantville NY 10570-9913</h5>

<h2>Education</h2>

<h3>Doctoral Students – Current</h3>

<h3>Students Completed Ph.D. / DPS</h3>

<h1 style="text-align: center">University Teaching – I<em>nauguration f</em><em>all 2001 </em><em>semester </em></h1>

<h2 style="text-align: left">Course Listing</h2>

<h4 style="text-align: left">(<strong><em>Course Materials in Blackboard</em></strong>):</h4>

<h4 style="text-align: left"><strong>Graduate Courses</strong><strong style="font-size: 19px">– <strong>Baruch College</strong></strong><strong>: </strong></h4>

<h4 style="text-align: left"><strong style="font-size: 19px">Undergraduate Courses – <strong>Baruch College</strong></strong><strong style="font-size: 19px">: </strong></h4>

<h4 style="text-align: left"><strong style="font-size: 19px">Undergraduate Courses – <strong>YORK College</strong></strong><strong style="font-size: 19px">: </strong></h4>

<h4 style="text-align: left"><strong style="font-size: 19px">Undergraduate Courses – <strong>LAGUARDIA Community College</strong></strong><strong style="font-size: 19px">: </strong></h4>

<h4><strong>CAPS Courses</strong><strong style="font-size: 19px">– <strong>Baruch College</strong></strong><strong>:</strong></h4>

<h4 style="text-align: left">COURSEs Teaching – by Semesters:</h4>

<h3><a href="http://www.baruch.cuny.edu/" rel="noopener noreferrer" target="_blank">Baruch College</a>/CUNY ( Fall 2006 to CURRENT )<br/>
<a href="https://cis.baruch.cuny.edu/">Computer Information Systems Dept </a></h3>

<h4>(<em>Course Materials in Blackboard</em>)</h4>

<h4 style="font-weight: 400;text-align: left"><a href="http://www.baruched.com/">Baruch Contiuning and Professional Studies (CAPS)</a></h4>

<h4>(<em>Course Materials in Moodle</em>):</h4>

<h4 style="font-weight: 400;text-align: left"><a href="http://www.york.cuny.edu/academics/departments/mathematics-and-computer-science">Math &amp; Computer Studies Dept.</a></h4>

<h4>(<em>Course Materials in Blackboard</em>)</h4>

<h4 style="text-align: left">Academic Computing and Educational Technology Dept.</h4>

<h4 style="text-align: left"><a href="https://www.york.cuny.edu/student-development/seek/">SEEK Dept</a></h4>

<h3 style="text-align: left"><a href="http://www.lagcc.cuny.edu/">LaGuardia Community College</a>/CUNY (Fall 2001 to Spring 2003 )</h3>

<h4 style="text-align: left">Computer Science Dept.<u style="font-weight: 400"><br/>
</u></h4>

<h2><strong>My CV/ Resume:</strong></h2>

<h2>Google Scholar Citations:</h2>

<h2>Github Repo:</h2>

<h2 class="widget-title">Recent Posts</h2>

<h2 class="widget-title">Recent Comments</h2>

<h2 class="widget-title">Archives</h2>

<h2 class="widget-title">Categories</h2>

<h2 class="widget-title">Meta</h2>

## Question#4:
### Complete the Python program below to extract and display all the image links from the URL below.



In [12]:
#Doesn't work anymore
from urllib.request import urlopen
from bs4 import BeautifulSoup
import re

#saves url data
html = urlopen('https://bctc.baruch.cuny.edu/')
#parses data to html
bs = BeautifulSoup(html, 'html.parser')

#saves all images with a source of .jpg that are tagges <img>
images = bs.find_all('img',{'src':re.compile('.jpg')})
#loop that prints out the image 'scr' links with a line space inbetween
for image in images: 
    print(image['src']+'\n')
    
#i didn't include the png images to have the same output that is provided.

URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)>

**OUTPUT TEXT:**

https://bctc.baruch.cuny.edu/wp-content/uploads/sites/26/2020/12/Welcome-to-BCTC.jpg

https://bctc.baruch.cuny.edu/wp-content/uploads/sites/26/2021/01/Sliders_Live-Suppor-Extra-Days.jpg

https://bctc.baruch.cuny.edu/wp-content/uploads/sites/26/2020/11/Sliders-Virtual-Computer-Labs.jpg

## Question#5:
### Complete the Python program below to list all language names and number of related articles in the order they appear in the URL below


In [6]:
from urllib.request import urlopen
from bs4 import BeautifulSoup

html = urlopen('https://www.wikipedia.org/')
bs = BeautifulSoup(html, "html.parser")

#all the languages and the associated amount of articles are stored under a with the class of link-box
#some the namelist stores all elements under <a> in the class of link-box
nameList = bs.find_all('a',{'class':'link-box'})

#loop that prints the name of each language and the number of articles connected to it
for name in nameList:
  print(name.get_text())





English
6 383 000+ articles


日本語
1 292 000+ 記事


Русский
1 756 000+ статей


Deutsch
2 617 000+ Artikel


Español
1 717 000+ artículos


Français
2 362 000+ articles


中文
1 231 000+ 條目


Italiano
1 718 000+ voci


Português
1 074 000+ artigos


Polski
1 490 000+ haseł



**OUTPUT TEXT:**


English
6 245 000+ articles


日本語
1 252 000+ 記事


Deutsch
2 534 000+ Artikel


Español
1 659 000+ artículos


Русский
1 697 000+ статей


Français
2 296 000+ articles


Italiano
1 672 000+ voci


中文
1 175 000+ 條目


Polski
1 454 000+ haseł


Português
1 055 000+ artigos

Polski
1 424 000+ haseł


## Question#6: (Optional)
### Write a Python program to get the number of followers of a given twitter account

In [ ]:
 
 from bs4 import BeautifulSoup
import requests

handle = input('Input your account name on Twitter: ')
temp = requests.get('https://twitter.com/'+handle)
bs = BeautifulSoup(temp.text,'lxml')

try:
    tweet_box = #TO DO -- Complete the Code
    tweets= #TO DO -- Complete the Code
    print("{} tweets {} number of tweets.".format(handle,tweets.get('data-count')))

except:
    print('Account name not found...')

**OUTPUT TEXT:**

Input your account name on Twitter: DrAbuKam
Account name not found...

## Reference(s) title & URL:
